source/common/http/http1/balsa_parser.cc

```diff
@@ -37,6 +37,12 @@ BalsaParser::BalsaParser(MessageType type, ParserCallbacks* connection, size_t m
 
 size_t BalsaParser::execute(const char* slice, int len) {
   ASSERT(status_ != ParserStatus::Error);
+
+  if (len == 0 && headers_processed_ && !isChunked() &&
+      ((!framer_.is_request() && hasTransferEncoding()) || !headers_.content_length_valid())) {
```

Comment:
Curious, does Http1ClientConnectionImplTest.NoContentLengthResponse set the Transfer-Encoding header somewhere?

Reply:
No, I don't think so.

Reply:
Hm, I am then a bit confused on how this code block comes into play for that test, but it clearly must! (Ah, maybe missing Content-Length is !headers_.content_length_valid()?)

Reply:
That's right.  `content_length_status_` starts out as `NO_CONTENT_LENGTH` [1], so if there is no Content-Length header, then content_length_valid() [2] returns false.

[1] https://sourcegraph.com/github.com/google/quiche@53c94d9/-/blob/quiche/balsa/balsa_headers.cc?L219
[2] https://sourcegraph.com/github.com/google/quiche@53c94d9/-/blob/quiche/balsa/balsa_headers.h?L942